modules/ztp-pre-staging-downloading-artifacts.adoc

```diff
@@ -0,0 +1,416 @@
+-rw-r--r--. 1 root root  164432422 Oct 31 15:24 ocp-v4.0-art-dev@sha256_fc4783b446c70df30b3120685254b40ce13ba6a2b0bf8fb1645f116cf6a392f1.tgz
+-rw-r--r--. 1 root root  306643814 Oct 31 15:11 troubleshoot@sha256_b86b8aea29a818a9c22944fd18243fa0347c7a2bf1ad8864113ff2bb2d8e0726.tgz
+----
+
+[id="ztp-downloading-operator-images_{context}"]
+== Downloading the Operator images
+
+You can also pre-cache Day 2 Operators used in the 5G Radio Access Network (RAN) Distributed Unit (DU) cluster configuration. The Day 2 Operators depend on the installed {product-title} version.
+
+[IMPORTANT]
+====
+You need to include the {rh-rhacm} hub version, so that the {factory-prestaging-tool} can pre-stage the appropriate containers images for the {rh-rhacm} and MCE Operators.
+====
+
+.Procedure
+
+. Pre-cache the Operator images:
++
+[source,terminal]
+----
+# podman run -v /mnt:/mnt -v /root/.docker:/root/.docker --privileged -it --rm quay.io/user/telco-ran-tools:acm-versioning -- factory-precaching-cli \
+    download -r 4.11.5 -f /mnt \
+    --ai-img registry.redhat.io/multicluster-engine/assisted-installer-agent-rhel8@sha256:da1753f9fcb9e229d0a68de03fac90d15023e647a8db531ae489eb93845d5306 \
+    --ai-img registry.redhat.io/multicluster-engine/assisted-installer-reporter-rhel8@sha256:e8d6b78248352b1a8e05a22308185a468d4a139682d997a7f968b329abbc02cd \
+    --ai-img registry.redhat.io/multicluster-engine/assisted-installer-rhel8@sha256:33abd6e21cfdc36dd4337fa6f3c3442d33fc3f976471614dca5b8ef749e7a027 \
+    --img quay.io/alosadag/troubleshoot \
+    --du-profile --acm-hub 2.5.4 -s <1>
+----
+<1> The `--du-profile` argument specifies pre-staging the Operators included in the DU configuration.
+
+.Example output
++
+[source,terminal]
+----
+Generated /mnt/imageset.yaml
+Generating list of pre-cached artifacts...
+Processing artifact [1/379]: ocp-v4.0-art-dev@sha256_7753a8d9dd5974be8c90649aadd7c914a3d8a1f1e016774c7ac7c9422e9f9958
+Processing artifact [2/379]: ose-kube-rbac-proxy@sha256_c27a7c01e5968aff16b6bb6670423f992d1a1de1a16e7e260d12908d3322431c
+Processing artifact [3/379]: ocp-v4.0-art-dev@sha256_370e47a14c798ca3f8707a38b28cfc28114f492bb35fe1112e55d1eb51022c99
+...
+Processing artifact [378/379]: ose-local-storage-operator@sha256_0c81c2b79f79307305e51ce9d3837657cf9ba5866194e464b4d1b299f85034d0
+Processing artifact [379/379]: multicluster-operators-channel-rhel8@sha256_c10f6bbb84fe36e05816e873a72188018856ad6aac6cc16271a1b3966f73ceb3
+----
+
+[id="ztp-custom-pre-caching-in-disconnected-environment_{context}"]
+== Pre-caching custom images in disconnected environments
+
+By default, the {factory-prestaging-tool} enables the argument `--generate-imageset`. This argument creates an `imageset` custom resource definition (CRD) including the {product-title} release and the optional DU Operators. The `imageset` CRD generates an `ImageSetConfiguration` custom resource (CR) based on the arguments passed to the tool.
```

Comment:
I should update this section of our doc, as well, as this isn't quite accurate. The `--generate-imageset` argument isn't enabled by default. Rather, it tells the tool to halt after generating the imageset config file, without performing any image downloads. In conjunction with the `--skip-imageset` option, this gives the user an opportunity to customize the imageset config file as desired. Essentially, these two options split the procedure into two separate stages.

Reply:
I rephrased it, PTAL:
> The `--generate-imageset` argument stops the {factory-prestaging-tool} after the `ImageSetConfiguration` custom resource (CR) is generated. This allows you to customize the `ImageSetConfiguration` CR before downloading any images. After you customized the CR, you can use the `--skip-imageset` argument to download the images that you specified in the `ImageSetConfiguration` CR.